src/coreclr/jit/codegenlinear.cpp

```diff
@@ -2484,8 +2484,8 @@ CodeGen::GenIntCastDesc::GenIntCastDesc(GenTreeCast* cast)
         }
 
 #if defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
-        // For LoongArch64's ISA which is same with the MIPS64 ISA, even the instructions of 32bits operation need
-        // the upper 32bits be sign-extended to 64 bits.
+        // For LoongArch64's ISA which is a RISC ISA which is different from any other existing ones,
+        // even the instructions of 32bits operation need the upper 32bits be sign-extended to 64 bits.
```

Comment:
Perhaps simplify the comment without drawing comparisons? If ifdef on line above is enough for reader to understand the platform context.
```suggestion
        // 32-bit operations sign-extend their results to 64 bits.
```

Reply:
This can be OK.

Reply:
Sorry but I still don't think it's correct.  The code logic is "if the input code performs a cast from 64-bit int to 32-bit int, we need to do a sign-extension."  But "32-bit operations sign-extend their results to 64 bits" does **not** explain the necessity of the sign-extend operation (i.e. it does **not** explain why we cannot just do a move like all other targets).

To me it should be something like

> On some 64-bit LoongArch CPU, the 32-bit division instructions need the inputs to be sign-extended to the GPR width; and on both 64-bit LoongArch and 64-bit RISC-V there's no 32-bit comparison instructions, so 32-bit values must be sign-extended to the GPR width before being compared.  Ensure all 32-bit values sign-extended to satisfy those requirements.

My word still does not explain why we cannot just do the sign-extension before dividing or comparing two 32-bit values, instead of (over-actively, to me) sign-extend all 32-bit values.

Reply:
Member

Reply:
2488